### DISCUSSION TITLE
Publish enhancement, Assigning elements.

### DISCUSSION THREAD

**Discussion by plzcloseyoureyes** [uNetworking/uWebSockets.js#378] [Q&A] [answered]
uWebSockets_App.publish(`users/id/${userID}`, 'message goes here', false, false, { stuff to assign });

example
```uWebSockets_App.publish(`users/id/${userID}`, 'message goes here', false, false, { hey: true, new_id: '12345' });```
and then it will do something like Object.assign does, to the socket object, and when accessed next it will be return the updated values.

This can be really useful for performance aswell if it will be coded in uWebSockets core, because at the moment in order to do such thing, you have to make an array containing all of the connections and update them through that array, by using .find() or something similiar.

That's just a suggestion that could be a perofrmance beast!

**Comment by ghost** [comment, marked as answer]
The purpose of publish is to publish messages over the affected connections. If you want to change values to objects that is out of scope and does not belong in that function.

You don't need to solve it the way you propose, you can easily solve it in O(log n) or even O(1) by using smarter algorithms in your coding. It doesn't belong in this library.
